### ISSUE TITLE
Can't install on a FP2 w/ TWRP

### ISSUE THREAD

**Issue by Mannshoch** [ubports/ubports-installer#52]
I followed the advice of UbPorts and now I'm at a death end. I got the Advice to reboot into Recovery. I'm now in it but nothing more happened.

Ubuntu 17.10 in VMware Workstation 14
Installed with snap
stable branch
Fairphone 2

**Comment by NeoTheThird** [comment]
Huh, interesting. It might be the VMs fault, though. What's your host OS? Can't that run the UBports Installer? I'm not sure how usb-access is forwarded to the vm in VMware.

One thing you could try is running `echo 0x2ae5 >> ~/.android/adb_usb.ini` in the terminal before installation. If that doesn't help and the installer gets stuck again, please run `adb devices` in the terminal, to see if the device is connected.

Unrelated, but please remember to install the rc (or devel if you're feeling feisty) channel, if you have a relatively new FP2 (recognizable by the white seperator between rear camera and led).

**Comment by NeoTheThird** [comment]
Was the device automatically detected, or did you have to select it manually?

**Comment by Mannshoch** [comment]
The VM had no adb installed
After Installing adb I got
```
List of devices attached
adb server version (39) doesn't match this client (36); killing...
*** Error in `adb': free(): invalid next size (fast): 0x000055acce4d7170 ***
ADB server didn't ACK
* failed to start daemon *
error: cannot connect to daemon
```

**Comment by Mannshoch** [comment]
A bit strange every thing. 
ubports seems to detect Fairphone 2 without Problem if I connect my phone while I'm in TWRP. Since I installed adb on Ubuntu  ubports is now also able to detect while I'm in Android.

I changed back to Fairphone OS and tried it again. Starting ubports from Android now works and had installed

**Comment by NeoTheThird** [comment]
Did you try my other suggestions from above?

**Comment by Mannshoch** [comment]
> One thing you could try is running echo 0x2ae5 >> ~/.android/adb_usb.ini in the terminal before installation. If that doesn't help and the installer gets stuck again, please run adb devices in the terminal, to see if the device is connected.
Was not needed.

**Comment by NeoTheThird** [comment]
> What's your host OS? Can't that run the UBports Installer? I'm not sure how usb-access is forwarded to the vm in VMware.

**Comment by NeoTheThird** [comment]
Oh, i'm sorry, i misread your comment.

> Starting ubports from Android now works and had installed

So the installation did not work with TWRP, but did with vanilla Fairphone OS? Interesting.

**Comment by charlie-ablett** [comment]
Hi, I've got a similar issue. I have TWRP installed as flash recovery and I can't get to a bootloader (the initial "Fairphone" screen just hangs forever).

Can I install manually please? I think that's the only way this will work.

**Comment by Ingo-FP-Angel** [comment]
> I can't get to a bootloader (the initial "Fairphone" screen just hangs forever).

That actually is the Fairphone 2 bootloader mode, it just shows the boot splash image, nothing else.
Manual install instructions can be found at https://devices.ubuntu-touch.io/device/FP2

**Comment by NeoTheThird** [comment]
This should be solved by #933 and will be included [in the next release](https://github.com/ubports/ubports-installer/releases/tag/0.4.1-beta)
